File: Shared/Blog.Contracts/V1/Requests/CommentsRequests/CreateCommentRequest.cs
namespace Blog.Contracts.V1.Requests.CommentsRequests;

using System.ComponentModel.DataAnnotations;
using Interfaces;

/// <summary>
/// Create comment request.
/// </summary>
public class CreateCommentRequest : IRequest
{
    /// <summary>
    /// Gets or sets the post identifier.
    /// </summary>
    /// <value>
    /// The post identifier.
    /// </value>
    [Required]
    public int PostId { get; set; }

    /// <summary>
    /// Gets or sets the comment body.
    /// </summary>
    /// <value>
    /// The comment body.
    /// </value>
    [Required]
    public string CommentBody { get; set; }

    /// <summary>
    /// Gets or sets the user identifier.
    /// </summary>
    /// <value>
    /// The user identifier.
    /// </value>
    public string UserId { get; set; }

    /// <summary>
    /// Gets or sets the name.
    /// </summary>
    /// <value>
    /// The name.
    /// </value>
    public string Name { get; set; }

    /// <summary>
    /// Gets or sets the email.
    /// </summary>
    /// <value>
    /// The email.
    /// </value>
    public string Email { get; set; }
}

File: Shared/Blog.FSharp.Contracts/V1/Requests/CommentsRequests/CreateCommentRequest.fs
namespace Blog.FSharp.Contracts.V1.Requests.CommentsRequests

open System.ComponentModel.DataAnnotations
open Blog.FSharp.Contracts.V1.Requests.Interfaces

/// <summary>
/// Create comment request.
/// </summary>
type CreateCommentRequest() =
    interface IRequest

    /// <summary>
    /// Gets or sets the post identifier.
    /// </summary>
    [<Required>]
    member val PostId: int = 0 with get, set

    /// <summary>
    /// Gets or sets the comment body.
    /// </summary>
    [<Required>]
    member val CommentBody: string = "" with get, set

    /// <summary>
    /// Gets or sets the user identifier.
    /// </summary>
    member val UserId: string = "" with get, set

    /// <summary>
    /// Gets or sets the name.
    /// </summary>
    member val Name: string = "" with get, set

    /// <summary>
    /// Gets or sets the email.
    /// </summary>
    member val Email: string = "" with get, set